dashboard/app/helpers/pd/workshop_survey_results_helper.rb

```diff
@@ -159,4 +159,66 @@ def summarize_workshop_surveys(workshops:, facilitator_name_filter: nil, facilit
 
     sum_hash
   end
+
+  def generate_workshop_daily_session_summary(workshop, related_workshops)
+    # TODO: (mehal) - Logic to only allow this for selected summer workshops
+    summary = {
+      this_workshop: {},
+      all_my_workshops: {},
+      all_workshops: {}
+    }
+    summary['questions'] = get_questions_for_form
+
+    workshop.sessions.each_with_index do |session, day|
+      summary[:this_workshop]["Day #{day + 1}"] = generate_sessions_survey_summary([session], include_free_response: true)
+    end
+
+    session_count = workshop.sessions.count
+    session_count.times do |session_index|
+      sessions = related_workshops.map {|related_workshop| related_workshop.sessions[session_index]}
+      summary[:all_my_workshops]["Day #{session_index + 1}"] = generate_sessions_survey_summary(sessions)
+    end
+
+    summary
+  end
+
+  def generate_sessions_survey_summary(sessions, include_free_response: false)
+    surveys = get_surveys_for_sessions(sessions)
+
+    questions = get_questions_for_form
+    session_summary = {
+      attending_teachers: sessions.map(&:attendances).map(&:count).reduce(:+),
+      completed_surveys: surveys.count
+    }
+    questions.each do |q_key, question|
+      if include_free_response && question[:free_response]
+        sum = surveys.map {|survey| survey[q_key]}.reduce([], :append)
+        session_summary[q_key] = sum
+      elsif question[:free_response].nil?
```

Comment:
why `nil?` specifically? What if we specify `free_response: false`?

Reply:
Restructured the conditionals.